src/extractor/guidance/intersection_handler.cpp

```diff
@@ -51,6 +51,11 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     const auto &in_data = node_based_graph.GetEdgeData(via_edge);
     const auto &out_data = node_based_graph.GetEdgeData(road.eid);
 
+    // suppress instructions on edges that the can't control anyway
+    if (!in_data.startpoint && !out_data.startpoint)
```

Comment:
I disagree here. This would mean that you need to add explicit handling based on mode for every mode you want to support. That means the probability that we need to find and adjust this piece of code in the future is very high, which I think is undesirable and duplicates logic from the profile. 

Using `startingpoint` uses a concept that we already expose in the profiles: Flagging ways that should be threaded as a single 'unit' from a user perspective. The name might be good or bad but this is effectively what this does. I don't see any scenario where we would reasonably emit an instruction between edges of this type. Any division between steps is done by the routing engine for technical reasons not user intend.

I feel this will be easier to use from a user perspective and offer less surprises then explicitly needing to cycle back every time a user finds a use-case for having `ferry`-like edges.